packages/changed-elements-react/src/NamedVersionSelector/NamedVersionSelector.tsx

```diff
@@ -137,15 +137,106 @@ export function NamedVersionSelectorWidget(props: NamedVersionSelectorWidgetProp
   );
 }
 
+function EmptyState(): ReactElement {
+  return (
+    <Text className="_cer_v1_empty-state" isMuted>
+      {t("VersionCompare:versionCompare.noPastNamedVersions")}
+    </Text>
+  );
+}
+
+function LoadingState(): ReactElement {
+  return (
+    <LoadingContent>
+      <Text>
+        {t("VersionCompare:versionCompare.loadingNamedVersions")}
+      </Text>
+    </LoadingContent>
+  );
+}
+
+type loadedStateProps = {
+  iTwinId: string;
+  iModelId: string;
+  currentNamedVersion: NamedVersion;
+  entries: NamedVersionEntry[];
+  onNamedVersionOpened: (version: NamedVersionEntry) => void;
+  updateJobStatus: ReturnType<typeof useNamedVersionsList>["updateJobStatus"];
+  emptyState?: ReactNode;
+  manageVersions?: ReactNode;
+};
+
+function LoadedState(props: Readonly<loadedStateProps>): ReactElement {
+  const {
+    iTwinId,
+    iModelId,
+    currentNamedVersion,
+    entries,
+    onNamedVersionOpened,
+    updateJobStatus,
+    emptyState,
+    manageVersions,
+  } = props;
+
+  return (
+    <NamedVersionSelectorLoaded
+      iTwinId={iTwinId}
+      iModelId={iModelId}
+      currentNamedVersion={currentNamedVersion}
+      entries={entries}
+      onNamedVersionOpened={onNamedVersionOpened}
+      updateJobStatus={updateJobStatus}
+      emptyState={emptyState}
+      manageVersions={manageVersions}
+    />
+  );
+}
+
+type renderContentArgs = {
```

Comment:
PascalCase, also consider changing `Args` to `Props` given that its used by a functional component